.github/workflows/go-mod.yaml

```diff
@@ -0,0 +1,25 @@
+name: Go module vendoring
+
+on:
+  pull_request:
+    paths:
+    - 'go.mod'
+    - 'go.sum'
+    - 'vendor/**'
```

Comment:
If we want to make this new workflow a requirement to merge pull request, the above filters are going to pose an issue: the workflow won't execute on pull requests that don't modify the above files and therefore, those pull requests will never become `status:success`.

We had this issue before, for the Documentation workflow.

Reply:
I discussed solutions with GitHub support, but right now, fixing this would require us to write a script to implement the same path filters ourselves :disappointed: We did not deem this worth it for the Documentation workflow since it was short.

Reply:
Yes, I think so. At some point we'll have so many short workflows that writing the script will be worthwhile :smile:

Reply:
Thanks. I changed the workflow to run on all PRs and also adjusted the commit message.